email_handler.py

```diff
@@ -927,28 +932,32 @@ def forward_email_to_mailbox(
 
     replace_sl_message_id_by_original_message_id(msg)
 
-    # change the from_header so the email comes from a reverse-alias
-    # replace the email part in from: header
-    old_from_header = msg[headers.FROM]
-    new_from_header = contact.new_addr()
-    add_or_replace_header(msg, "From", new_from_header)
-    LOG.d("From header, new:%s, old:%s", new_from_header, old_from_header)
-
-    if reply_to_contact:
-        reply_to_header = msg[headers.REPLY_TO]
-        new_reply_to_header = reply_to_contact.new_addr()
-        add_or_replace_header(msg, "Reply-To", new_reply_to_header)
-        LOG.d("Reply-To header, new:%s, old:%s", new_reply_to_header, reply_to_header)
+    # Don't Change Headers when SMTP is enabled for alias
+    if not is_SMTP_enabled_for_alias:
```

Comment:
Won't VERP handle this ? SPF record are checked against the envelope sender and not the `From:` header, right ?

The reason I added this was to prevent confusion and threading issue in the mail client.

Say user is using Gmail's "Send Mail as:" and send a email to a contact. 
Then, if this is not there, then while getting reply, the `From:` will be be a reverse-alias. It will cause the email to not be in the thread and might create confusion for the user.

Reply:
You're right, the email will pass SPF. The email doesn't have envelope and header alignment and will fail DMARC though. 

The threading is handled by the headers "Message-ID", "In-Reply-To" and "References". 

To handle the case where reverse alias and contact address are mixed, I think we can handle the case the envelope.rcpt_to is a reverse alias in SMTP handler. In this case we can consider that the same as envelope.rcpt_to is the contact address. Let me know if that makes sense :).

Reply:
> You're right, the email will pass SPF. The email doesn't have envelope and header alignment and will fail DMARC though.

Yes, for DMARC to pass the original DKIM signature should be preserved. 

--
> The threading is handled by the headers "Message-ID", "In-Reply-To" and "References".

Oh ok ok
--

> To handle the case where reverse alias and contact address are mixed, I think we can handle the case the envelope.rcpt_to is a reverse alias in SMTP handler. In this case we can consider that the same as envelope.rcpt_to is the contact address. Let me know if that makes sense :).

 I did not get this part. Can you please explain this ?

Reply:
> I did not get this part. Can you please explain this ?

Currently SMTP handler receives an email from an alias to a contact and send the email to the contact.

What I meant is in addition to that flow, SMTP handler can also handle the case user sends an email from an alias to a reverse alias. So when SMTP handler receives an email from an alias to a reverse alias, it extracts the contact address from the reverse alias and continue the normal process.

Reply:
> So you are saying that, the SMTP handler should handle this case as well, right?

Yep.

> Also, in this scenario, the reverse-alias, should belong to the alias, correct ?

Yep.

This case can happen when:

1) user sends an email to a contact via SMTP (so no reverse alias)
2) contact replies, the response will be sent from the reverse alias
3) user replies back, the reply will be sent from the alias to the reverse alias -> it's where SMTP handler is called so we need to handle this case.

Reply:
> 2. contact replies, the response will be sent from the reverse alias
 
I certainly believe this would cause confusion to the users. 
We can just forward the email as-is without changing the `From:` to reverse-alias.

I get that you are concerned that it will fail DMARC, but I think all we need to do is preserve the original DKIM signature, then add our DKIM signature (which is already being done), the DMARC will be passed.
 I guess that is how the pure forwarding services (like ImprovMX, Cloudflare Email Routing, etc) do it.

We can also look into adding the ARC set headers if the emails are still ending up in spam.

Reply:
As I mentioned earlier, the email will fail dmarc alignement so we can't keep the original header. We **cannot** keep original header at the moment. This code part needs to be reverted in order for the PR to be merged.

ARC is another topic that should probably be addressed in another PR.

Reply:
I have reverted this part.